CharlieBackend.Api/Controllers/ExportController.cs

```diff
@@ -56,14 +84,19 @@ public async Task<IActionResult> GetStudentsClassbook([FromBody] StudentsRequest
         [Authorize(Roles = "Mentor, Secretary, Admin")]
         [Route("studentsResults")]
         [HttpPost]
-        public async Task<IActionResult> GetStudentsResults([FromBody] StudentsRequestDto<StudentResultType> request)
+        public async Task<IActionResult> GetStudentsResults([FromBody] StudentsRequestWithFileExtensionDto<StudentResultType> request)
         {
-            var studentResults = await _exportService.GetStudentsResults(request);
+            CheckExtension(request.Extension);
 
-            return File(await
-                studentResults.GetByteArrayAsync(),
-                studentResults.GetContentType(),
-                studentResults.GetFileName());
+            var results = await _dashboardService
+                .GetStudentsResultAsync(request.GetStudentsRequestDto());
+
```

Comment:
Do not initialize _exportService in constructor. When it is needed, just get it from factory. var exportService = _exportFactoryService.GetExportService(request.Extension);